serde_arrow/benches/groups/json_to_arrow.rs

```diff
@@ -0,0 +1,101 @@
+use crate::groups::complex_common::Item;
+
+use {
+    serde_arrow::schema::{SchemaLike as _, SerdeArrowSchema},
+    std::sync::Arc,
+};
+
+// arrow-version:replace: use arrow_json_{version}::ReaderBuilder;
+use arrow_json_50::ReaderBuilder;
+
+// arrow-version:replace: use arrow_schema_{version}::Schema as ArrowSchema;
+use arrow_schema_50::Schema as ArrowSchema;
+
+// arrow-version:replace: use arrow_array_{version}::RecordBatch;
+use arrow_array_50::RecordBatch;
+
+fn benchmark_json_to_arrow(c: &mut criterion::Criterion) {
+    let rng = &mut rand::thread_rng();
+    let items = (0..10_000)
+        .map(|_| Item::random(rng))
+        .collect::<Vec<Item>>();
+    let jsons_to_deserialize = items
+        .iter()
+        .map(|item| serde_json::to_string(item).expect("Failed to serialize JSON"))
+        .collect::<Vec<_>>();
+    let jsons_to_deserialize_concatenated = jsons_to_deserialize.join("\n");
+    let jsons_to_deserialize: Vec<&str> = {
```

Comment:
Maybe it's fairer to use the same string as input to both and include the lines in the benchmark, i.e.,

```
json_to_deserialize.lines().for_each(...)
```

Probably needs a check for empty lines (see my impl below)

Reply:
added my benchmark code to this effect